lte/gateway/c/session_manager/SessionState.cpp

```diff
@@ -1530,4 +1542,18 @@ void SessionState::update_bearer_deletion_req(
   update.delete_req.mutable_eps_bearer_ids()->Add(
       bearer_id_by_policy_[PolicyID(policy_type, rule_id)]);
 }
+
+void SessionState::update_data_usage_metrics(
+    uint64_t bytes_tx, uint64_t bytes_rx) {
+  auto sid    = get_config().common_context.sid().id();
+  auto msisdn = get_config().common_context.msisdn();
+  auto apn    = get_config().common_context.apn();
+  increment_counter(
+      "ue_reported_tx", bytes_tx, size_t(3), LABEL_IMSI, sid.c_str(), LABEL_APN,
+      apn.c_str(), LABEL_MSISDN, msisdn.c_str());
+  increment_counter(
+      "ue_reported_rx", bytes_rx, size_t(3), LABEL_IMSI, sid.c_str(), LABEL_APN,
+      apn.c_str(), LABEL_MSISDN, msisdn.c_str());
+}
```

Comment:
What's the difference between these and the `ue_traffic` counter which already has labels for direction? Per-UE timeseries are extremely expensive to store and query on Prometheus at any sort of scale and we should avoid adding more unless absolutely necessary

Reply:
`ue_traffic` use case is more for tracking and monitoring through a specific rating group and policy rule, these two are meant to cover session data usage for both LTE and CWF use cases (CWF currently gets UL/DL throughput metrics through the AP which seems to be a weak method). After we uniform these and confirm it works for both for our use cases then we can look to remove the ones in radius / CWF side.

Reply:
Can we feature-gate this to CWF only then? Will we be maintaining 2 sets of UE data usage timeseries in FWA if we don't?

I want to stress that it's *extremely* expensive to store and query these timeseries in Prometheus, particularly if label cardinality is high.

Reply:
Ideally we should be using the same session metrics for FWA and CWF. But yeah I understand the concern on scalability and performance, I think we can start with these for CWF but in long term we might want to use these for both.

Reply:
Ideally we should be using the same session metrics for FWA and CWF. But yeah I understand the concern on scalability and performance, I think we can start with these for CWF but in long term we might want to use these for both. Tagging @amarpad for his input

Reply:
Session counters per APN is also a specific request from Entel so we need something for FWA as well.

My higher level question here is should we have this be an event associated with session termination vs a counter.

Reply:
Is that a separate conversation, or the same thing? You mean report data usage only as part of a session termination event? It's slightly different semantics right? One is updated periodically (metric) and the other only set once per termination event. Also one will be queryable in grafana and the other not.

@ardzoht Let's use a single metric name for this and make direction tx/rx a label.

Reply:
@xjtian Added a single metric and included the direction as label name